File: clients/java/examples/collections-list-collection-metadata.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.definition.CollectionDescriptor;
import com.datastax.astra.client.databases.Database;
import java.util.List;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    // List collections
    List<CollectionDescriptor> collectionMetadata = database.listCollections();
    collectionMetadata.stream().map(CollectionDescriptor::getOptions).forEach(System.out::println);
  }
}

File: clients/csharp/examples/collections-list-collection-metadata.cs
using System.Text.Json;
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get a database
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );

    // List collections
    var collectionMetadata = await database.ListCollectionsAsync();

    Console.WriteLine(JsonSerializer.Serialize(collectionMetadata));
  }
}
